### PR TITLE
Angular 1.6.x

### PR DESCRIPTION
**Pull request by paulojreis**
This PR upgrades Angular to version 1.6 (latest).

### CODE REVIEW

File: app/views/layout.pug [hunk 1]
```diff
@@ -68,7 +68,7 @@ html(itemscope, itemtype='http://schema.org/Product')
 
 		block scripts
 
-		script(src=buildJsPath("libs/angular-1.3.15.min.js", {fingerprint:false}))
+		script(src=buildJsPath("libs/angular-1.6.4.js", {fingerprint:false}))
```
Comment (jpallen) [inline app/views/layout.pug]: Should we use the min version?

File: public/coffee/directives/asyncForm.coffee [hunk 2]
```diff
@@ -51,14 +52,15 @@ define [
...
+						.catch (httpResponse) ->
+							{ data, status, headers, config } = httpResponse
 							scope[attrs.name].inflight = false
 							response.success = false
 							response.error = true
 
 							onErrorHandler = scope[attrs.onError]
 							if onErrorHandler
-								onErrorHandler(data, status, headers, config)
+								onErrorHandler(httpResponse)
```
Comment (ShaneKilkelly) [inline public/coffee/directives/asyncForm.coffee]: Just a note for anyone reviewing, but the only place these `onErrorHandler` and `onSuccessHandler` are used, is in the `launchpad` module.

File: public/coffee/ide/chat/services/chatMessages.coffee [hunk 3]
```diff
@@ -47,7 +47,8 @@ define [
 			chat.state.loading = true
 			return $http
 				.get(url)
-				.success (messages = [])->
+				.then (response) ->
+					messages = response.data ? []
```
Comment (ShaneKilkelly) [inline public/coffee/ide/chat/services/chatMessages.coffee]: Very minor stylistic comment (no need to change anything), we've tended to use `x || y` in cases of conditional assignment, but I do like the `?` operator better. Maybe we should use this more in future.

File: public/coffee/ide/history/controllers/HistoryDiffController.coffee [hunk 4]
```diff
@@ -9,7 +9,7 @@ define [
 				.restoreDeletedDoc(
 					$scope.history.diff.doc
 				)
-				.success (response) ->
+				.then (response) ->
```
Comment (ShaneKilkelly) [inline public/coffee/ide/history/controllers/HistoryDiffController.coffee]: I suspect this should be followed by a `{data} = response`, then adjust line 13 to be `... = data.doc_id`

File: app/views/layout.pug (hunk 1 above)
Comment (paulojreis) [inline app/views/layout.pug]: Sure. This was just for easier debugging.

**Comment by ShaneKilkelly** [comment]
LGTM

**Pull request merged**